src/sympc/store/crypto_primitive_provider.py

```diff
@@ -13,6 +14,8 @@ class CryptoPrimitiveProvider:
     """A trusted third party should use this class to generate crypto primitives."""
 
     _func_providers: Dict[str, Callable] = {}
+    _logging = False
+    _ops_list = {}
```

Comment:
MAYBE: Make it a ```defaultdict``` and on line 54 you won't need to check if it has the element

Reply:
done